src/main/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessageSendingQueueProcessor.java

```diff
@@ -66,38 +66,44 @@ public void start() {
             e.printStackTrace();
         }
-
-
     }
 
     private void attemptSend(WebhookDeliveryQueueEntity queueItem) {
         var retryCount = webhookDeliveryQueueDao.countFailed(queueItem.getWebhookMessageEntity());
-        
+
         var webhookMessageSender = new WebhookMessageSender(httpClient, objectMapper, webhookMessageSignatureGenerator);
         try {
             LOGGER.info("Attempting to send Webhook ID %s to %s".formatted(queueItem.getWebhookMessageEntity().getExternalId(), queueItem.getWebhookMessageEntity().getWebhookEntity().getCallbackUrl()));
             var response = webhookMessageSender.sendWebhookMessage(queueItem.getWebhookMessageEntity());
+
             var statusCode = response.statusCode();
             if (statusCode >= 200 && statusCode <= 299) {
+                LOGGER.info("Message attempt succeeded with %s".formatted(statusCode));
                 webhookDeliveryQueueDao.recordResult(queueItem, String.valueOf(statusCode), statusCode, WebhookDeliveryQueueEntity.DeliveryStatus.SUCCESSFUL);
             } else {
+                LOGGER.info("Message attempt failed with %s".formatted(statusCode));
                 webhookDeliveryQueueDao.recordResult(queueItem, String.valueOf(statusCode), statusCode, WebhookDeliveryQueueEntity.DeliveryStatus.FAILED);
                 enqueueRetry(queueItem, nextRetryIn(retryCount));
             }
         } catch (HttpTimeoutException e) {
+            LOGGER.info("HTTP timeout exception %s".formatted(e.toString()));
             webhookDeliveryQueueDao.recordResult(queueItem, "HTTP Timeout after 5 seconds", null, WebhookDeliveryQueueEntity.DeliveryStatus.FAILED);
             enqueueRetry(queueItem, nextRetryIn(retryCount));
         } catch (IOException | InterruptedException | InvalidKeyException e) {
-            LOGGER.warn("Unexpected exception %s attempting to send webhook message ID: %s".formatted(e.getMessage(), queueItem.getWebhookMessageEntity().getExternalId()));
+            LOGGER.warn("Exception %s attempting to send webhook message ID: %s".formatted(e.getMessage(), queueItem.getWebhookMessageEntity().getExternalId()));
             webhookDeliveryQueueDao.recordResult(queueItem, e.getMessage(), null, WebhookDeliveryQueueEntity.DeliveryStatus.FAILED);
             enqueueRetry(queueItem, nextRetryIn(retryCount));
-            
+        } catch (Exception e) {
+            // handle all exceptions at this level to make sure that the retry mechanism is allowed to work as designed
+            // allowing errors passed this point (not guaranteeing an update) would allow perpetual failures 
+            LOGGER.warn("Unexpected exception %s attempting to send webhook message ID: %s".formatted(e.getMessage(), queueItem.getWebhookMessageEntity().getExternalId()));
+            webhookDeliveryQueueDao.recordResult(queueItem, "Unknown error", null, WebhookDeliveryQueueEntity.DeliveryStatus.FAILED);
```

Comment:
Maybe it would be better to propagate the exception message up here? On the other hand I guess no guarantee it would be ledgeble to an external user.

Reply:
Hmm yeah I had considered this but am a bit concerned about exposing system level messages out to consumers without some kind of map. I can't think how but it feels like there could be some sharp edges or unintended behaviour. Not totally against it, it will be good to refine these delivery status. messages once the pieces are in place.